pmlc/dialect/pxa/ir/ops.td

```diff
@@ -57,11 +57,29 @@ def AffineParallelOp : PXA_Op<"parallel", [ImplicitAffineTerminator]> {
   let extraClassDeclaration = [{
     operand_range getLowerBoundsOperands();
     operand_range getUpperBoundsOperands();
+
+    // Get the number of dimensions (i.e. number of induction variables)    
+    size_t getNumDims();
+
+    // Get various AffineValueMaps
+    AffineValueMap getLowerBoundsValueMap(); 
+    AffineValueMap getUpperBoundsValueMap(); 
+    AffineValueMap getRangesValueMap();
+
+    // Get ranges as constants, return true if it worked (non-dynamic case)
+    bool getConstantRanges(llvm::SmallVectorImpl<int64_t>& out);
     
+    // For a given dimension, check if range is constant.  If so, return, otherwise return -1
+    int64_t getRange(size_t dim);
```

Comment:
Probably use `unsigned` instead of `size_t`

Reply:
Actually, this has no implementation, just deleted it.